### PR TITLE
Add autoupdate workflow for dapps

### PR DESCRIPTION
**Pull request by nmattia**
This adds a daily workflow run that checks
https://github.com/dfinity/portal and updates our local list of dapps & logos.

This involves a few changes:

* A new script, `scripts/update-dapps`, is added and used by the workflow.
* This script was run and the dapps are up-to-date.
* The explorer & other components making use of logos are updated to only prefix the logo with BASE_URL (& URL path prefix) where the logos are used (the dapps list only gives the logo name now, for clarity & brevity)
* The vite build is updated to avoid compressing SVG assets (the canister chokes on those, and due to the updated dapps data, we now have SVG assets)

The vite build is updated to _not_ compress svgs, since

### CODE REVIEW

File: src/frontend/src/flows/authorize/index.ts [hunk 1]
```diff
@@ -71,13 +72,13 @@ export const authnTemplateAuthorize = ({
     <div class="c-origin-preview c-origin-preview--header">
       <img
         class="c-origin-preview__logo c-origin-preview__logo--background"
-        src=${logo}
+        src=${BASE_URL + "icons/" + logo}
```
Comment (nmattia) [inline src/frontend/src/flows/authorize/index.ts:75]: The `BASE_URL + "icons/"` pattern is used in a couple of places, and ideally should be defined only once. I'll tackle this when adding support for preloading those images.

**Comment by frederikrothenberger** [comment]
I think the script should have a size cut-off to only include a dapp if the logo is below a certain size (e.g. <10KB).
Currently we have the top 8 largest icons account for over 70% of the `icons` size. Without such a limit we will soon hit the wasm size limit.

**Review by frederikrothenberger — APPROVED** [review]
LGTM, apart from the file size issue 👍 
Thanks!

**Comment by nmattia** [comment]
@frederikrothenberger I've added two steps, one that tries to optimize logos (currently turns big SVG into raster image) and one that drops logos if any are still too big. Here's the result:

```
found big logo 'src/frontend/assets/icons/cosmicrafts_logo.webp'
found big logo 'src/frontend/assets/icons/hot_or_not_logo.svg'
converting 'src/frontend/assets/icons/hot_or_not_logo.svg' -> 'src/frontend/assets/icons/hot_or_not_logo.webp'
found big logo 'src/frontend/assets/icons/factland_logo.png'
found logo still too big 'src/frontend/assets/icons/cosmicrafts_logo.webp', removing
found logo still too big 'src/frontend/assets/icons/factland_logo.png', removing
```

https://github.com/dfinity/internet-identity/actions/runs/5077713740

**Comment by frederikrothenberger** [comment]
Nice, this made the wasm significantly smaller 👍

File: src/frontend/src/flows/authorize/index.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/flows/authorize/index.ts:75]: https://github.com/dfinity/internet-identity/pull/1678

**Pull request merged**